src/neo/SmartContract/Native/Tokens/NeoToken.cs

```diff
@@ -131,11 +154,29 @@ protected override void PostPersist(ApplicationEngine engine)
 
             // Distribute GAS for committee
 
-            int index = (int)(engine.Snapshot.PersistingBlock.Index % (uint)ProtocolSettings.Default.CommitteeMembersCount);
+            int m = ProtocolSettings.Default.CommitteeMembersCount;
+            int n = ProtocolSettings.Default.ValidatorsCount;
+            int index = (int)(engine.Snapshot.PersistingBlock.Index % (uint)m);
             var gasPerBlock = GetGasPerBlock(engine.Snapshot);
-            var pubkey = GetCommittee(engine.Snapshot)[index];
+            var committee = GetCommittee(engine.Snapshot);
+            var pubkey = committee[index];
             var account = Contract.CreateSignatureRedeemScript(pubkey).ToScriptHash();
             GAS.Mint(engine, account, gasPerBlock * CommitteeRewardRatio / 100);
+
+            // Record the cumulative reward of the voters of committee
+
+            var factor = index < n ? 2 : 1;
+            CandidateState candidate = engine.Snapshot.Storages.TryGet(CreateStorageKey(Prefix_Candidate).Add(pubkey))?.GetInteroperable<CandidateState>();
+            if (candidate?.Votes > 0)
+            {
+                BigInteger voterSumRewardPerNEO = factor * gasPerBlock * VoterRewardRatio * 100000000L * m / (m + n) / 100 / candidate.Votes; // Zoom in 100000000 times, and the final calculation should be divided 100000000L
```

Comment:
Why this formula?

Reply:
_1) The voters who voted for a consensus node are assigned 2/28 of this part according to the voting weight, 
2) and the voters who voted for a committee member (non-consensus node) are assigned 1/28 of this part according to the voting weight._ 

**Case1: one block:**
```js
m = the number of committee
n = the number of validator

VoterRewards = gasPerBlock * 85%
The voter reward of committee = VoterRewards / 28 = gasPerBlock * 85% / (m + n)
The voter reward of validator = VoterRewards / 28 * 2 = gasPerBlock * 85% * 2 / (m + n)
 ```

**Case 2: one epoch**
 ( = 21 block, = the number of committee)
```js
The voter reward of committee = VoterRewards / 28 * Epoch = gasPerBlock * 85% / (m + n) * m
The voter reward of validator = VoterRewards / 28 * 2 * Epoch = gasPerBlock * 85% * 2 / (m + n) * m
```